### ISSUE TITLE
Nix not installing on Monteray 12.4

### ISSUE THREAD

**Issue by addy** [NixOS/nix#6744]
**Describe the bug**

Getting the following error when running the `sh <(curl -L https://nixos.org/nix/install)` command:

```sh
---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo install -dv -m 0755 /nix /nix/var /nix/var/log /nix/var/log/nix /nix/var/log/nix/drvs /nix/var/nix /nix/var/nix/db /nix/var/nix/gcroots /nix/var/nix/profiles /nix/var/nix/temproots /nix/var/nix/userpool /nix/var/nix/daemon-socket /nix/var/nix/gcroots/per-user /nix/var/nix/profiles/per-user

to make the basic directory structure of Nix (part 1)

install: chmod 755 /nix: Read-only file system
install: mkdir /nix/var: Read-only file system

---- oh no! --------------------------------------------------------------------
Jeeze, something went wrong. If you can take all the output and open
an issue, we'd love to fix the problem so nobody else has this issue.
```

Full log: https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941

**Steps To Reproduce**

- For Bash
  - Run `sh <(curl -L https://nixos.org/nix/install)`
- For Fish
  - Run `sh (curl -L https://nixos.org/nix/install | psub)`

1. Select option `n`
2. Select option `y`
3. Select option `y`
4. Type sudo password and hit enter

**Expected behavior**

Nix is installed

**Additional context**

I have tried uninstalling following: https://nixos.org/manual/nix/stable/installation/installing-binary.html#macos

I still get the same error each time, regardless of the current shell I'm using.

**Comment by abathur** [comment]
Have you had Nix on this system before? If so, when would you guess you first installed it?

Can you run the following and report output?
- `stat /nix`
- `cat /etc/fstab`
- `cat /etc/synthetic.conf`

**Comment by addy** [comment]
This is a relatively fresh system, no Nix installs before this one.

Here's the output for those four commands: https://gist.github.com/addy/d5ce07c5c4d6ea92ab9e4d73cc985827

**Comment by abathur** [comment]
The read-only errors mean your nix store volume isn't getting mounted, but I don't see any strong hint as to why. 

One of these steps might be silently failing:
- https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941#file-output-txt-L214
- https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941#file-output-txt-L222

You can also check the plist file in the first command to try out the actual mount command. Maybe worth confirming the uuid there, in fstab, and in the output of `diskutil apfs list` all match up.

**Comment by addy** [comment]
Sorry for the late reply, I checked the outputs of those various programs and confirmed that the UUID matches across all of them.

I do have FileVault enabled on this computer, which I cannot disable. I did, however, see that it was in an Unlocked state, so that might be another indication that the above commands worked successfully.

Is this line of the original output more of a side effect of the above, or could it be another issue that's happening with mounting? https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941#file-output-txt-L2071

**Comment by abathur** [comment]
> I checked the outputs of those various programs and confirmed that the UUID matches across all of them.
👍 

> I do have FileVault enabled on this computer, 

FileVault shouldn't be a problem (though its setup is a little more complicated). 

> which I cannot disable. 

I assume you mean this is an org device? It is _possible_ that something in your organization's profiles are interfering. (But if so, they're doing it in a way we haven't heard about yet. #6669 reports what would likely be in the output if your profile prevented mounting. The next-most-obvious thing might be if they were restricting use of the system keychain or `security` command. The credential insertion step at https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941#file-output-txt-L172-L190 appears to have gone fine, but I guess it might be blocking read access?)

> I did, however, see that it was in an Unlocked state, so that might be another indication that the above commands worked successfully.

Yes, (with the asterisk that it's harder to be confident about what was true immediately after the installer ran, if you manually ran the launchctl commands) I _think_ the unmount step at https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941#file-output-txt-L199 put the volume in a locked state, suggesting that it was unlocked by either of the steps at https://gist.github.com/addy/fc6160d49d4eeb0b34f2932a1b5c3941#file-output-txt-L211-L224

(This would likely rule out the question of whether a profile is blocking the keychain use.)

I guess I'd try:
1. reboot
2. Check whether it's mounted with `diskutil info /nix`. (It'll error if not, or print a lot of information if so.)

    - If the volume _is_ mounted, it's a sign the volume got set up correctly during the installer and that something else is causing trouble. You could 
    - Assuming the volume is _not_ mounted, run `cat /Library/LaunchDaemons/org.nixos.darwin-store.plist`, extract the unlock/mount command from the plist file, and manually run it. Each argument will be wrapped in a `<string>` tag, so it might be easier to just use that copy as a reference to populate this template: 

        ```console
        /usr/bin/security find-generic-password -s '$volume_uuid' -w | /usr/sbin/diskutil apfs unlockVolume '$volume_uuid' -mountpoint '$mountpoint' -stdinpassphrase 
        ```

**Comment by addy** [comment]
> I assume you mean this is an org device? 
Yes, this is an org device, but I don't see that specific SUIS related error in the logs.

I just rebooted and checked the `/nix` mount. Seems like it mounted correctly and is unencrypted. I see that corresponding key in the key store, so that makes sense. When running the command in `/Library/LaunchDaemons/org.nixos.darwin-store.plist` I get the following:

`The APFS Volume disk1s7 is already unlocked (mounted)`

I see some people have similar errors in other GitHub issues, but those all seem to have been resolved by this point. It feels like my Mac isn't respecting the synthetic mount for whatever reason. Maybe it is due to some strange org restrictions.

**Comment by addy** [comment]
To finally loop back on this, looks like Richard found the solution specific to our situation on #9070
